docs/eventing/sources/containersource.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "ContainerSource"
+linkTitle: "ContainerSource"
+weight: 31
+type: "docs"
+---
+
+![version](https://img.shields.io/badge/API_Version-v1beta1-red?style=flat-square)
+
+ContainerSource will start a container image which will generate events under
+certain situations and send messages to a sink URI. It also can be an easy way
+to support your own event sources in Knative. This guide shows how to configure
+ContainerSource as an event source for functions and summarizes guidelines for
+creating your own event source as a ContainerSource.
+
+### Prerequisites
+- Install [ko](https://github.com/google/ko)
+- Set [`KO_DOCKER_REPO`](https://github.com/knative/serving/blob/master/DEVELOPMENT.md#environment-setup)
+ (e.g. `gcr.io/[gcloud-project]` or `docker.io/<username>`)
+- Authenticated with your `KO_DOCKER_REPO`
+- Install [`docker`](https://docs.docker.com/install/)
+
+## Installation
+
+The ContainerSource source type is enabled by default when you install Knative Eventing.
+
+## Example
+
+This example shows how the heartbeats container sends events to a Knative Service.
+
+### Prepare the heartbeats image
```

Comment:
I would remove this section and the the published heartbeats image. BUT since this image is not published yet, we can't. 

At the very least, can you remove the dependency on Knative serving?

Reply:
Thanks for the comment! I have 2 confusions.
1. If we don't ko publish the image and as you mentioned, there is no public image for the heartbeats image. What image path should we use in the container to create the source?
2. Could you please be more specific about what it means by `remove the dependency on Knative Serving`? Like remove the pre-requisites or ...?

Thanks again!

Reply:
created an issue for the heartbeats image: https://github.com/knative/eventing-contrib/issues/1418

Reply:
Should we keep our PR on hold until there is a published image available? Or we can go back to fix it after they publish it.

Reply:
Sorry I think I'm still confused about what you meant by `remove the dependency on Knative serving`?

Reply:
see other inlined comments, thanks!